subsys/bluetooth/shell/bt.c

```diff
@@ -994,6 +1003,64 @@ static int cmd_auto_conn(int argc, char *argv[])
 	return 0;
 }
 
+static int cmd_directed_adv(int argc, char *argv[])
+{
+	int err;
+	bt_addr_le_t addr;
+	struct bt_le_adv_param *param = BT_LE_ADV_CONN_DIR;
+
+	if (argc < 2) {
+		return -EINVAL;
+	}
+
+	if (!strcmp(argv[1], "off")) {
+		err = bt_conn_disconnect(default_conn, 0);
+		if (err < 0) {
+			printk("Failed to stop directed advertising: %d\n",
+			       err);
+		} else {
+			bt_conn_unref(default_conn);
+			default_conn = NULL;
+			printk("Directed advertising stopped\n");
+		}
+
+		return 0;
+	} else if (!strcmp(argv[1], "on")) {
+		if (argc < 4) {
+			return -EINVAL;
+		}
+	} else {
+		return -EINVAL;
+	}
+
+	err = str2bt_addr_le(argv[2], argv[3], &addr);
+	if (err) {
+		printk("Invalid peer address (err %d)\n", err);
+		return 0;
+	}
+
+	if (argc > 4) {
+		if (!strcmp(argv[4], "low")) {
+			param = BT_LE_ADV_CONN_DIR_LOW_DUTY;
+		} else {
+			return -EINVAL;
+		}
+	}
+
+	if (default_conn) {
+		printk("Pointer to the connection instance is not free\n");
```

Comment:
Would it not make sense to return an error here, otherwise you create a memory leak of a bt_conn object?

Reply:
Yes, that's true - I forgot about it. However, I suggest to just return 0 at this point, because then the user won't get this command usage information (the command is correct in this case, but its not possible to start advertising because of application state).